### ISSUE TITLE
Handling detection of NPM package binaries in scripts

### ISSUE THREAD

**Issue by denizdogan** [webpro/knip#221]
I have a custom NPM script which uses `nodemon`, `ts-node`, and `pino-pretty` to watch and run a TypeScript module.

Knip does not detect the usage of any of these packages. What is the intended way to handle this in Knip?

This is my package.json, basically:

```json
{
  "scripts": {
    "example:simple": "nodemon --exec 'ts-node --esm' ./example/simple.ts | pino-pretty --singleLine",
    "knip": "knip"
  },
  "devDependencies": {
    "knip": "^2.19.5",
    "nodemon": "^3.0.1",
    "ts-node": "^10.9.1"
  },
  "dependencies": {
    "pino-pretty": "^10.2.0"
  }
}
```

This is my knip.json:

```json
{
  "entry": ["src/index.ts", "example/simple.ts"],
  "project": ["src/**/*.ts", "example/**/*.ts"]
}
```

Output:

```console
$ pnpm knip

Unused dependencies (1)
pino-pretty  package.json
Unused devDependencies (2)
nodemon  package.json
ts-node  package.json
 ELIFECYCLE  Command failed with exit code 3.
```

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.21.0. See [Release 2.21.0](https://github.com/webpro/knip/releases/tag/2.21.0) for release notes._

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.21.0. See [Release 2.21.0](https://github.com/webpro/knip/releases/tag/2.21.0) for release notes._

**Comment by webpro** [comment]
Thanks @denizdogan, piping was not supported yet, same for nodemon's `exec` argument (so two fixes here).
